### DISCUSSION TITLE
Simplify geojson

### DISCUSSION THREAD

**Discussion by mikima** [d3/d3#3584] [Q&A] [answered]
Hi all!

I'm drawing a path from a geojson file.
I'd like to simplify it on the flight according to the different zoom level to avoid too heavy svgs, therefore reducing the number of points.

I've seen that there is a ["precision" option for projections](https://github.com/d3/d3-geo#projection_precision), but if i got it well it's more for adding points rather than removing ones.

I've also seen that there is a [simplify-js library](http://mourner.github.io/simplify-js/), but I found examples with topojson but not with geojson.

Do you have any suggestion? thank you!

**Comment by Fil** [comment, marked as answer]
The gold standard is still this block: https://bl.ocks.org/mbostock/6287633 (be sure to follow the links to see variants).
It uses an old version of D3, but the general principle still holds.

**Reply by Fil** [reply]
Here's a version that is updated to current versions of D3 and topojson
https://observablehq.com/@fil/dynamic-simplification

let me know if you have questions!

**Reply by mikima** [reply]
Hi! thank you for the answer and the example. I'll have a look at them asap. I see you use a topojson, is it possible to do it also with a geojson?

**Reply by Fil** [reply]
You need topojson to do simplification.

The data flow for this notebook loads us.json as topojson, then converts it to geojson to project it to screen coordinates, then topojson again to add weights to each of the points, then geojson again to use with geoPath… :-)

**Reply by mikima** [reply]
thank you, I forked your observable testing it with SVG: https://observablehq.com/@mikima/dynamic-simplification-svg

My overall goal was to make it smooth the interaction when dealing with complex shapefiles. I'd like to test also the usage of the simplify-js library.
